src/Orckestra.Composer/Providers/CurrencyProvider.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Orckestra.Composer.Services;
+using System;
+
+namespace Orckestra.Composer.Providers
+{
+    public class CurrencyProvider: ICurrencyProvider
+    {
+        protected IComposerContext ComposerContext { get; private set; }
+
+        public CurrencyProvider(IComposerContext composerContext)
+        {
+            ComposerContext = composerContext ?? throw new ArgumentNullException(nameof(composerContext));
+        }
+
+        public virtual string GetCurrency()
```

Comment:
There's no reason to make this method "virtual". If a custom implementation needed, a developer can implement a different ICurrencyProvider, instead of inheriting this one.